File: app/src/main/java/com/android/ashwiask/tvmaze/AppComponent.java
package com.android.ashwiask.tvmaze;

import com.android.ashwiask.tvmaze.home.HomeComponent;
import com.android.ashwiask.tvmaze.home.HomeModule;
import com.android.ashwiask.tvmaze.network.NetworkModule;

import javax.inject.Singleton;

import dagger.Component;

/**
 * @author Ashwini Kumar.
 */
@Singleton
@Component(modules = {AppModule.class, NetworkModule.class})
public interface AppComponent
{
    HomeComponent plus(HomeModule homeModule);
}

File: app/src/main/java/com/android/tvmaze/di/AppComponent.kt
package com.android.tvmaze.di

import android.app.Application
import com.android.tvmaze.TvMazeApplication
import com.android.tvmaze.db.TvMazeDbModule
import com.android.tvmaze.network.NetworkModule
import dagger.BindsInstance
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(
    modules = [AndroidSupportInjectionModule::class,
        AppModule::class,
        ActivityBuildersModule::class,
        NetworkModule::class,
        TvMazeDbModule::class]
)
interface AppComponent {
    fun inject(tvMazeApplication: TvMazeApplication)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }
}
